README.md

```diff
@@ -92,27 +104,133 @@ Returns an array of objects:
 [
   {
     "name": "vite",
+    "engines": {
+      "node": "^18.0.0 || >=20.0.0"
+    },
     "specifier": "alpha",
-    "version": "6.0.0-alpha.18",
-    "lastSynced": 1719798752827
+    "version": "6.0.0-alpha.24",
+    "publishedAt": "2024-09-04T09:15:20.820Z",
+    "lastSynced": 1735496799048
   },
   {
     "name": "vue",
     "specifier": "latest",
-    "version": "3.4.31",
-    "lastSynced": 1719799051285
+    "version": "3.5.13",
+    "publishedAt": "2024-11-15T14:48:51.738Z",
+    "lastSynced": 1735496870654
   },
   {
     "name": "nuxt",
+    "engines": {
+      "node": "^14.18.0 || >=16.10.0"
+    },
     "specifier": "~3.11",
     "version": "3.11.2",
-    "lastSynced": 1719799051232
+    "publishedAt": "2024-04-04T16:10:14.223Z",
+    "lastSynced": 1735496870614
   }
 ]
 ```
 
 ### `GET /versions/:pkg`
 
+#### ❓Options
+
+You can pass options with query parameters.
+
+<details>
+<summary>
+<h5> Engines: Add engines for each versions</h5>
+</summary>
+
+###### Append (_default_)
+
+As it is the default value you can use the shorthand and don't set a value
+
+```sh
+curl https://npm.antfu.dev/versions/vite?engines
+curl https://npm.antfu.dev/versions/vite?engines=append
+```
+
+```jsonc
+{
+  "name": "vite",
+  "distTags": {
+    "alpha": "6.0.0-alpha.24",
+    "previous": "5.1.8",
+    "beta": "6.0.0-beta.10",
+    "latest": "6.0.6"
+  },
+  "versions": [
+    "0.1.0",
+    "0.1.1",
+    "0.1.2"
+    // ...
+  ],
+  "versionsEngines": {
+    "0.6.1": {
+      "node": ">=10.0.0"
+    },
+    "0.7.0": {
+      "node": ">=10.0.0"
+    },
+    "0.8.0": {
+      "node": ">=10.0.0"
+    }
+    // ...
+  },
+  "time": {
+    "created": "2020-04-21T05:05:15.476Z",
+    "modified": "2024-12-26T02:23:38.890Z",
+    "0.1.0": "2020-04-21T05:05:15.591Z",
+    "0.1.1": "2020-04-21T05:06:51.876Z"
+    // ...
+  },
+  "specifier": "*",
+  "lastSynced": 1735407539388
+}
+```
+
+###### Concat
+
+```sh
+curl https://npm.antfu.dev/versions/vite?engines=concat
+```
+
+```jsonc
+{
+  "name": "vite",
+  "distTags": {
+    "alpha": "6.0.0-alpha.24",
+    "previous": "5.1.8",
+    "beta": "6.0.0-beta.10",
+    "latest": "6.0.6"
+  },
+  "versions": {
+    "0.6.1": {
+      "node": ">=10.0.0"
+    },
```

Comment:
I wonder when the `concat` mode is useful? 

I am not even sure if this format makes sense, as it seems a bit confusing to have the node range directly in `versions`. And also a bit worried that this would make future extension of adding new fields harder. I think the `versionsEngines` mode should be good enough, where we could also simplify the `engines` query to a boolean.